### DISCUSSION TITLE
Real-time reading of sensors data

### DISCUSSION THREAD

**Discussion by RinnetenseiQ** [zepp-health/zeppos-forum#110] [Developer Feedback]
Is there any way to read f.e. raw acceleration data from watch (i have amazfit gtr 4) in real-time? Is it possible to get it directly using BLE or from Zepp app? I want to use raw sensors data in my scientific researching.

**Comment by RinnetenseiQ** [comment]
I have tried to find watch using BLE on Python:

```
import asyncio
from bleak import BleakScanner


async def run():
    devices = await BleakScanner.discover()
    print(devices)
    for d in devices:
        print(d)


if __name__ == '__main__':
    loop = asyncio.get_event_loop()
    loop.run_until_complete(run())
```
But I had no success:
```
C:\ProgramData\Anaconda3\envs\amazfit\python.exe C:/ITProjects/Pycharm/amazfit/conn.py
[]

Process finished with exit code 0
```

Even I my PC thinks that watch are connected now:
![image](https://user-images.githubusercontent.com/49791012/224487041-2d089cc3-4bc4-448d-97ed-10f29321bc99.png)

On watch in in some watchscreen that shows bluetoth status it is looks like watch is not connected:
![image](https://user-images.githubusercontent.com/49791012/224487118-a7c705f8-8ef8-423f-a624-06fb916f7926.png)

**Comment by zdunex25** [comment]
All available sensors are listed here: https://docs.zepp.com/docs/reference/device-app-api/newAPI/sensor/Battery/

**Reply by RinnetenseiQ** [reply]
As I understand, there is no way to get acceleration data?

**Reply by zdunex25** [reply]
Not in zeppos 2.0, maybe in the future, but who knows.

**Reply by RinnetenseiQ** [reply]
Thx. As I understood,  Amazfit device purchase for researching is not a good idea

**Comment by MyoungXUE** [comment]
Hi, the acceleration sensor is in the planning of Zepp OS 3.0, please refer to https://github.com/orgs/zepp-health/discussions/59 for details.

Your idea of getting data through BLE is a very good idea, we are working on the technical problems and will open it at the right time.
